rocAL/rocAL_pybind/examples/rocAL_api_coco_pipeline.py

```diff
@@ -45,123 +24,82 @@ class ROCALCOCOIterator(object):
-          encodded_labels_tensor = torch.tensor(self.encoded_labels).long().view(self.bs, -1)
         else:
-          torch_gpu_device = torch.device('cuda', self.device_id)
-          boxes_array, labels_array = self.loader.getEncodedBoxesAndLables(self.bs, int(self.num_anchors))
-          self.encoded_bboxes = torch.as_tensor(boxes_array, dtype=torch.float32, device=torch_gpu_device)
-          self.encoded_labels = torch.as_tensor(labels_array, dtype=torch.int32, device=torch_gpu_device)
-          encoded_bboxes_tensor = self.encoded_bboxes.cpu()
-          encodded_labels_tensor = self.encoded_labels.cpu()
+            self.output_tensor_list = self.loader.rocalGetOutputTensors()
 
-        image_id_tensor = torch.tensor(self.image_id)
-        image_size_tensor = torch.tensor(self.img_size).view(-1, self.bs, 2)
-        for i in range(self.bs):
-            index_list = []
-            actual_bboxes = []
-            actual_labels = []
-            for idx, x in enumerate(encodded_labels_tensor[i]):
-                if x != 0:
-                    index_list.append(idx)
-                    actual_bboxes.append(encoded_bboxes_tensor[i][idx].tolist())
-                    actual_labels.append(encodded_labels_tensor[i][idx].tolist())
+        self.w = self.output_tensor_list[0].batch_width()
+        self.h = self.output_tensor_list[0].batch_height()
+        self.bs = self.output_tensor_list[0].batch_size()
+        self.color_format = self.output_tensor_list[0].color_format()
 
-            if self.display:
-                img = self.out
-                draw_patches(img[i], self.image_id[i],
-                             actual_bboxes, self.device)
+        torch_gpu_device = torch.device('cuda', self.device_id)
+
+        if self.tensor_format == types.NCHW:
+            self.out = torch.empty((self.bs, self.color_format, self.h, self.w,), dtype=torch.float32, device=torch_gpu_device)
+        else:
+            self.out = torch.empty((self.bs, self.h, self.w, self.color_format,), dtype=torch.float32, device=torch_gpu_device)
+        self.output_tensor_list[0].copy_data(ctypes.c_void_p(self.out.data_ptr()))
+
+        labels_array = self.loader.rocalGetBoundingBoxLabel()
+        encodded_labels_tensor = []
+        encoded_bboxes_tensor = []
+        for label in labels_array:
+            self.encoded_labels = torch.as_tensor(label, dtype=torch.int64)
+            encodded_labels_tensor.append(self.encoded_labels)
+
+        boxes_array = self.loader.rocalGetBoundingBoxCords()
+        for box in boxes_array:
+            self.encoded_bboxes = torch.as_tensor(box, dtype=torch.float16)
+            self.encoded_bboxes = self.encoded_bboxes * 800
+            self.encoded_bboxes = self.encoded_bboxes.view(-1, 4)
+            encoded_bboxes_tensor.append(self.encoded_bboxes)
+
+        matched_idxs = self.loader.rocalGetMatchedIndices()
```

Comment:
Need not include MatchedIndices in the example we can issue a separate PR for the match